src/preferences/dialog/dlgprefeffects.cpp

```diff
@@ -12,53 +13,61 @@ DlgPrefEffects::DlgPrefEffects(QWidget* pParent,
           m_pEffectsManager(pEffectsManager) {
     setupUi(this);
 
-    connect(availableEffectsList,
-            SIGNAL(currentItemChanged(QListWidgetItem*, QListWidgetItem*)),
+    m_pAvailableEffectsModel = new EffectSettingsModel();
+
+    m_pAvailableEffectsModel->resetFromEffectManager(pEffectsManager);
+    availableEffectsList->setModel(m_pAvailableEffectsModel);
+
+    connect(availableEffectsList->selectionModel(),
+            SIGNAL(currentRowChanged(const QModelIndex&, const QModelIndex&)),
             this,
-            SLOT(slotEffectSelected(QListWidgetItem*, QListWidgetItem*)));
+            SLOT(availableEffectsListItemSelected(const QModelIndex&)));
+
+    // Highlight first row
+    availableEffectsList->selectRow(0);
+
+    availableEffectsList->setColumnWidth(0, 50);
+}
+
+DlgPrefEffects::~DlgPrefEffects() {
+    delete m_pAvailableEffectsModel;
 }
 
 void DlgPrefEffects::slotUpdate() {
     clear();
-    const QList<EffectManifestPointer> availableEffectManifests =
-            m_pEffectsManager->getAvailableEffectManifests();
-
-    for (const auto& pManifest : availableEffectManifests) {
-        QListWidgetItem* pItem = new QListWidgetItem();
-        pItem->setData(Qt::UserRole, pManifest->id());
-        pItem->setText(pManifest->displayName());
-        availableEffectsList->addItem(pItem);
-    }
+    m_pAvailableEffectsModel->resetFromEffectManager(m_pEffectsManager);
 
-    if (!availableEffectManifests.isEmpty()) {
-        availableEffectsList->setCurrentRow(0);
+    if (!m_pAvailableEffectsModel->isEmpty()) {
+        availableEffectsList->selectRow(0);
     }
 }
 
 void DlgPrefEffects::slotApply() {
-    // Nothing to apply.
+    for (EffectProfilePtr profile : m_pAvailableEffectsModel->profiles()) {
+        EffectManifest* pManifest = profile->getManifest();
+        pManifest->setVisibility(profile->isVisible());
+        m_pConfig->set(ConfigKey("[Visible Effects]", pManifest->id()), ConfigValue(profile->isVisible()));
```

Comment:
I was initially thinking of using a comma or semicolon separated list for the hidden effects, but this approach has simpler code and is more robust in case the effect name has the delimiter in it. :+1: